src/ast/passes/semantic_analyser.cpp

```diff
@@ -1483,7 +1483,7 @@ void SemanticAnalyser::visit(Map &map)
         map.vargs->at(i) = cast;
         expr = cast;
       }
-      else if (expr->type.IsCtxAccess())
+      else if (expr->type.IsPtrTy() && expr->type.IsCtxAccess())
```

Comment:
Indeed. In fact, only now the original comment underneath matches the condition:
```cpp
        else if (expr->type.IsPtrTy() && expr->type.IsCtxAccess())
        // map functions only accepts a pointer to a element in the stack
```
The check is really about pointers and `args` is not a pointer anymore, so I wouldn't state it explicitly. We can wrap this into some helper like `isNonStackPointer` but I don't think that it would help much.